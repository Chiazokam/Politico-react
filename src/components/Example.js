import React from "react";
import { connect } from 'react-redux';
import { increment, decrement } from '../actions/example';

class Example extends React.Component {

  increment = () => {
    this.props.increment();
  };

  decrement = () => {
    this.props.decrement();
  };

  render() {
    return (
      <div className="counter">
        <h2>Counter</h2>
        <div>
          <button onClick={this.decrement}>-</button>
          <span className="count">{this.props.count}</span>
          <button onClick={this.increment}>+</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => { 
  return {
    count: state.example.count };
  }

const mapDispatchToProps = {
  increment,
  decrement,
};
    
export default connect(mapStateToProps, mapDispatchToProps)(Example);
