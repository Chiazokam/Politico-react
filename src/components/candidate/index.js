import { BecomeCandidate } from './BecomeCandidate';
import { CreateCandidate } from './CreateCandidate'

export { BecomeCandidate, CreateCandidate };
