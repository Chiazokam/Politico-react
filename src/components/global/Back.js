import React, { Component } from 'react';

class Back extends Component {
  constructor(props) {
    super(props)
    this.goBack = this.goBack.bind(this);
  }
  goBack () {
    this.props.history.goBack();
  }
  render() {
    console.log(this.props)
    return (
      <div>
        {this.goBack}
      </div>
    )
  }
}

export default Back;
