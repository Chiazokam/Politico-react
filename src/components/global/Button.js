import React, {Component} from 'react';
import '../../styles/global/button.scss'

class Button extends Component {
  render() {
    return (
      <React.Fragment>
        <a>
          <button type="button" className={this.props.className}>
            {this.props.children}
          </button>
        </a>
      </React.Fragment>
    )
  }
}

export default Button;
