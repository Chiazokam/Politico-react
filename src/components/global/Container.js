import React from 'react';
import '../../styles/global/container.scss'

const Container = props => {
  return (
    <div className="container">
      {props.children}
    </div>
  )
}

export default Container;
