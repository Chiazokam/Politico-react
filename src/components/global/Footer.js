import React from 'react';
import '../../styles/global/footer.scss';

const Footer = () => {
  return (
      <footer className="home-footer">
        <div className="container">
          <div className="footer-text">
            <div className="footer-contact">
              <a className="footer-header">Contact Us</a>
              <a className="footer-contact-text"><i className="fas fa-phone"></i>07032425466</a>
              <a className="footer-contact-text"><i className="fas fa-home"></i>12B Folawiyo Bankole, Street</a>
            </div>
          </div>
            <p>P<span className="brandname-span">olitico</span>, copyright &copy; 2019</p>
        </div>
      </footer>
    )
}

export default Footer;
