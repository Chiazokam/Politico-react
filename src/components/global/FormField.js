import React from 'react';

const FormField = props => {
  return (
    <div className={props.className}>
      <div className='error'>{props.error}</div>
      <input
        type={props.type}
        name={props.name}
        onChange={(e) => props.onChange(e)}
        value={props.value}
        id={props.id}
        onFocus={(e) => props.onFocus(e)}
        placeholder={props.placeholder} required>
      </input>
    </div>
  )
}

export default FormField;
