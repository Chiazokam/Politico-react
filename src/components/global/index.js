import Button from './Button';
import Footer from './Footer';
import NavList from './NavList';
import Header from './Header';
import TextStrip from './TextStrip';
import PlainStrip from './PlainStrip';
import Container from './Container';
import FormField from './FormField';
import Url from './helper';
import Logout from './Logout';
import PartyCard from './Party-card';
import OfficeCard from './Office-card';
import ToastMessage from './ToastMessage';

export {
  Header,
  Button,
  Footer,
  NavList,
  TextStrip,
  PlainStrip,
  Container,
  FormField,
  Url,
  Logout,
  PartyCard,
  OfficeCard,
  ToastMessage
};
