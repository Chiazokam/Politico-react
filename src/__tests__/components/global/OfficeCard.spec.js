import React from 'react';
import { shallow } from 'enzyme';
import { OfficeCard } from '../../../components/global';

describe('Office Card', () => {
  test('should render Office Card Container correctly', () => {
    const wrapper = shallow(<OfficeCard />);

    expect(wrapper).toMatchSnapshot();
  });
});
