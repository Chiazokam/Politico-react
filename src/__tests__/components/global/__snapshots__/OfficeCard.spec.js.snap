// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Office Card should render Office Card Container correctly 1`] = `ShallowWrapper {}`;
